pinot-common/src/main/java/org/apache/pinot/common/partition/ReplicaGroupPartitionAssignmentGenerator.java

```diff
@@ -99,7 +103,11 @@ public ReplicaGroupPartitionAssignment buildReplicaGroupPartitionAssignment(Stri
       numColumnPartitions =
           tableConfig.getIndexingConfig().getSegmentPartitionConfig().getNumPartitions(partitionColumn);
     }
-    int numReplicas = tableConfig.getValidationConfig().getReplicationNumber();
+
+    TableType tableType = tableConfig.getTableType();
+    int numReplicas =
```

Comment:
The code in PinotTableRestletResource.ensureMinReplicas() handles the case of HLC and LLC correctly to get the number of replicas for realtime. Not sure if that code can be pulled up to some common area (since it includes some verify flags as well), but it will be good if we do so.

I suppose there is some way this code is not enabled for HLC?

Reply:
Tried to pull the code into a common Utils.
Fixed such that this code will not be run in HLC case.